network/src/protocols/rpc/mod.rs

```diff
@@ -102,18 +103,20 @@ pub struct InboundRpcRequest {
 }
 
 /// A wrapper struct for an outbound rpc request and its associated context.
-#[derive(Debug)]
+#[derive(Debug, Serialize)]
```

Comment:
do we benefit much with this one :P

Reply:
It gives us the protocol, and the duration, so I guess not really.

I think this is harder to determine what should and shouldn't be serialized